### PR TITLE
show AIO password

### PR DESCRIPTION
**Pull request by Zoey2936**
Would look like this:
grafik

### CODE REVIEW

**Comment by szaimen** [comment]
Actually I think this is probably more confusion than simply going with the show password button. So lets indeed use implementation from https://www.w3schools.com/howto/howto_js_toggle_password.asp

**Comment by Zoey2936** [comment]
> Actually I think this is probably more confusing than simply going with the show password checkbox. So lets just use implementation from https://www.w3schools.com/howto/howto_js_toggle_password.asp

this was just an alternative Idea, do it the way you want

**Comment by szaimen** [comment]
Would you mind adjusting the PR?

**Comment by Zoey2936** [comment]
in which way?

**Comment by szaimen** [comment]
To use the solution mentioned in https://www.w3schools.com/howto/howto_js_toggle_password.asp

**Comment by szaimen** [comment]
> or another idea: what about setting it to a password field and when the first letter/number/symbol ist filled in change it to a text field?

If this works, probably it would be the best solution. Thanks for the idea! :)

**Comment by Zoey2936** [comment]
I will try it later

**Comment by Zoey2936** [comment]
works now

File: php/public/forms.js [hunk 1]
```diff
@@ -39,6 +39,15 @@
     document.getElementById('overlay').classList.remove('loading');
   }
 
+  function showPassword() {
+    var Password = document.getElementById("Password");
+    if (Password.value !== "" && Password.type === "password") {
+      Password.type = "text";
+    } else {
```
Comment (Zoey2936) [inline php/public/forms.js:46]: maybe and else if would be better

File: php/public/forms.js [hunk 2]
```diff
@@ -39,6 +39,15 @@
     document.getElementById('overlay').classList.remove('loading');
   }
 
+  function showPassword() {
+    var Password = document.getElementById("Password");
+    if (Password.value !== "" && Password.type === "password") {
```
Comment (Zoey2936) [inline php/public/forms.js:44]:
```suggestion
    if (Password.value !== "" && Password.type !== "text") {
```

File: php/public/forms.js (hunk 1 above)
Comment (Zoey2936) [reply php/public/forms.js:46]:
```suggestion
    } else if (Password.value === "" && Password.type !== "password") {
```

**Comment by Zoey2936** [comment]
maybe do the same here:
https://github.com/nextcloud/all-in-one/blob/d0cae686365409daab32c71b962a7debd8e5262c/php/templates/containers.twig#L512
https://github.com/nextcloud/all-in-one/blob/d0cae686365409daab32c71b962a7debd8e5262c/php/templates/containers.twig#L5123

**Comment by Zoey2936** [comment]
@szaimen what do you think, should I also add this to the password change field?

File: php/public/forms.js [hunk 3]
```diff
@@ -39,6 +39,15 @@
...
+      Password.type = "text";
+    } else {
+      Password.type = "password";
+    }
+  }
```
Comment (szaimen) [inline php/public/forms.js:49]:
```suggestion
function showPassword(id) {
    let passwordField = document.getElementById(id);
    if (passwordField) {
        if (passwordField .value === "") {
          passwordField .type = "password";
        } else {
            if (passwordField .type === "password") {
                passwordField .type = "text";
            }
        }
    }
}
```

File: php/templates/login.twig [hunk 4]
```diff
@@ -8,7 +8,7 @@
             {% if is_login_allowed == true %}
                 <p>Log in using your Nextcloud AIO password:</p>
                 <form method="POST" action="/api/auth/login" class="xhr">
-                    <input type="text" autocomplete="off" name="password" placeholder="Password" />
+                    <input type="password" autocomplete="current-password" name="password" placeholder="Password" wfd-id="id0" id="Password" oninput="showPassword()">
```
Comment (szaimen) [inline php/templates/login.twig:11]:
```suggestion
                    <input type="passwordInput" autocomplete="current-password" name="password" placeholder="Password" id="Password" oninput="showPassword('passwordInput')">

```

**Comment by szaimen** [comment]
> @szaimen what do you think, should I also add this to the password change field?

Yes, can you also use the same function in all of the fields 

> @szaimen what do you think, should I also add this to the password change field?

yes, probably makes sense 👍

File: php/public/forms.js (hunk 3 above)
Comment (Zoey2936) [inline php/public/forms.js:49]:
```suggestion
  function showPassword(id) {
    let passwordField = document.getElementById(id);
    if (passwordField.value !== "" && passwordField.type !== "text") {
      passwordField.type = "text";
    } else if (passwordField.value === "" && passwordField.type !== "password") {
      passwordField.type = "password";
    }
  }
```

File: php/templates/login.twig (hunk 4 above)
Comment (Zoey2936) [reply php/templates/login.twig:11]: why do you change the type? you need to change the id

Comment (Zoey2936) [reply php/templates/login.twig:11]:
```suggestion
                    <input type="password" autocomplete="current-password" name="password" placeholder="Password" id="Password" oninput="showPassword('Password')">

```

Comment (szaimen) [reply php/templates/login.twig:11]:
```suggestion
                    <input type="password" autocomplete="current-password" name="password" placeholder="Password" 
                    id="passwordInput" oninput="showPassword('passwordInput')">

```

Comment (Zoey2936) [reply php/templates/login.twig:11]:
```suggestion
                    <input type="password" autocomplete="current-password" name="password" placeholder="Password" id="passwordInput" oninput="showPassword('passwordInput')">

```

File: php/public/forms.js [hunk 5]
```diff
@@ -39,6 +39,15 @@
     document.getElementById('overlay').classList.remove('loading');
   }
 
+  function showPassword(id) {
+    let passwordField = document.getElementById(id);
+    if (passwordField.value !== "" && passwordField.type !== "text") {
```
Comment (szaimen) [inline php/public/forms.js:44]:
```suggestion
    if (passwordField.type === "password" && passwordField.value !== "") {
```

File: php/public/forms.js [hunk 6]
```diff
@@ -39,6 +39,15 @@
...
+      passwordField.type = "text";
+    } else if (passwordField.value === "" && passwordField.type !== "password") {
```
Comment (szaimen) [inline php/public/forms.js:46]:
```suggestion
    } else if (passwordField.type === "text" && passwordField.value === "") {
```

File: php/templates/login.twig [hunk 7]
```diff
@@ -8,7 +8,8 @@
             {% if is_login_allowed == true %}
                 <p>Log in using your Nextcloud AIO password:</p>
                 <form method="POST" action="/api/auth/login" class="xhr">
-                    <input type="text" autocomplete="off" name="password" placeholder="Password" />
+                    <input type="password" autocomplete="current-password" name="password" placeholder="Password" id="passwordInput" oninput="showPassword('passwordInput')">
```
Comment (szaimen) [inline php/templates/login.twig:11]:
```suggestion
                    <input type="password" autocomplete="current-password" name="password" placeholder="Password" id="master-password" oninput="showPassword('master-password')">

```

File: php/templates/login.twig [hunk 8]
```diff
@@ -8,7 +8,8 @@
             {% if is_login_allowed == true %}
                 <p>Log in using your Nextcloud AIO password:</p>
                 <form method="POST" action="/api/auth/login" class="xhr">
-                    <input type="text" autocomplete="off" name="password" placeholder="Password" />
+                    <input type="password" autocomplete="current-password" name="password" placeholder="Password" id="passwordInput" oninput="showPassword('passwordInput')">
+
```
Comment (szaimen) [inline php/templates/login.twig:12]:
```suggestion
```

File: php/public/forms.js (hunk 5 above)
Comment (Zoey2936) [reply php/public/forms.js:44]: and why?

Comment (szaimen) [reply php/public/forms.js:44]: because it is more performant and I think the logic is more understandable by so

File: php/public/forms.js (hunk 6 above)
Comment (szaimen) [reply php/public/forms.js:46]: why?

**Review by szaimen — APPROVED** [review]
LGTM, thank you! :)

**Comment by szaimen** [comment]
@Zoey2936 I fear I get this issue: 
image

**Comment by Zoey2936** [comment]
sorry, can't check this at the moment

**Pull request merged**